src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/CaseOfficerRespondentEvidenceSubmittedPersonalisation.java

```diff
@@ -34,20 +34,8 @@ public String getTemplateId() {
 
     @Override
     public String getEmailAddress(AsylumCase asylumCase) {
-        final HearingCentre hearingCentre =
-                asylumCase
-                        .read(HEARING_CENTRE, HearingCentre.class)
-                        .orElseThrow(() -> new IllegalStateException("hearingCentre is not present"));
-
-        final String hearingCentreEmailAddress =
-                hearingCentreEmailAddresses
-                        .get(hearingCentre);
-
-        if (null == hearingCentreEmailAddress) {
-            throw new IllegalStateException("Hearing centre email address not found: " + hearingCentre.toString());
-        }
 
-        return hearingCentreEmailAddress;
+        return new EmailAddressFinder(hearingCentreEmailAddresses).getEmailAddress(asylumCase);
```

Comment:
You could inject `EmailAddressFinder` as a `Bean` in `CaseOfficer*Personalisation`constructor instead of creating new instance everytime. Sorry for late comment

Reply:
I talked with Tarun. It is fixed in next notification story. I approve PR conditionally ;)